src/Classifier.php

```diff
@@ -59,28 +64,42 @@ public function withAttribute(string ...$attributes): self
         return $new;
     }
 
+    /**
+     * @return iterable<class-string>
+     */
     public function find(): iterable
     {
         $countInterfaces = count($this->interfaces);
         $countAttributes = count($this->attributes);
 
-        if ($countInterfaces === 0 && $countAttributes === 0 && $this->parent === null) {
+        if ($countInterfaces === 0 && $countAttributes === 0 && $this->parentClass === null) {
             return [];
         }
 
         $this->scanFiles();
 
         $classesToFind = get_declared_classes();
+        $isWindows = DIRECTORY_SEPARATOR === '\\';
+        $directories = $this->directories;
 
-        $baseDirectory = $this->directory;
-        if (DIRECTORY_SEPARATOR === '\\') {
-            $baseDirectory = str_replace('/', '\\', $baseDirectory);
+        if ($isWindows) {
+            /** @var string[] $directories */
+            $directories = str_replace('/', '\\', $directories);
         }
 
         foreach ($classesToFind as $className) {
             $reflection = new ReflectionClass($className);
-            $filePath = $reflection->getFileName();
-            if ($filePath === false || !str_starts_with($filePath, $baseDirectory)) {
+
+            if (!$reflection->isUserDefined()) {
```

Comment:
Directory is required for search classes? If yes need modify constructor and make one parameter required.